panels/info/cc-info-panel.c

```diff
@@ -484,117 +479,59 @@ get_os_description (void)
 }
 
 static void
-query_done (GFile        *file,
-            GAsyncResult *res,
-            CcInfoPanel  *self)
+get_primary_disc_info (CcInfoPanel *self)
 {
-  GFileInfo *info;
-  GError *error = NULL;
-
-  info = g_file_query_filesystem_info_finish (file, res, &error);
-  if (info != NULL)
-    {
-      self->priv->total_bytes += g_file_info_get_attribute_uint64 (info, G_FILE_ATTRIBUTE_FILESYSTEM_SIZE);
-      g_object_unref (info);
-    }
-  else
-    {
-      if (g_error_matches (error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
-        {
-          g_error_free (error);
-          return;
-        }
-      else
-        {
-          char *path;
-          path = g_file_get_path (file);
-          g_warning ("Failed to get filesystem free space for '%s': %s", path, error->message);
-          g_free (path);
-          g_error_free (error);
-        }
-    }
-
-  /* And onto the next element */
-  get_primary_disc_info_start (self);
-}
+  GDBusObjectManager *manager;
+  GHashTable *added_drives;
+  GtkWidget *label;
+  GList *objects, *l;
+  guint64 total_size;
 
-static void
-get_primary_disc_info_start (CcInfoPanel *self)
-{
-  GUnixMountEntry *mount;
-  GFile *file;
+  label = WID ("disk_label");
+  total_size = 0;
 
-  if (self->priv->primary_mounts == NULL)
+  if (!self->priv->client)
     {
-      char *size;
-      GtkWidget *widget;
-
-      size = g_format_size (self->priv->total_bytes);
-      widget = WID ("disk_label");
-      gtk_label_set_text (GTK_LABEL (widget), size);
-      g_free (size);
-
+      gtk_label_set_text (GTK_LABEL (label), "-");
       return;
     }
 
-  mount = self->priv->primary_mounts->data;
-  self->priv->primary_mounts = g_list_remove (self->priv->primary_mounts, mount);
-  file = g_file_new_for_path (g_unix_mount_get_mount_path (mount));
-  g_unix_mount_free (mount);
-
-  g_file_query_filesystem_info_async (file,
-                                      G_FILE_ATTRIBUTE_FILESYSTEM_SIZE,
-                                      0,
-                                      self->priv->cancellable,
-                                      (GAsyncReadyCallback) query_done,
-                                      self);
-  g_object_unref (file);
-}
-
-static void
-get_primary_disc_info (CcInfoPanel *self)
-{
-  GList        *points;
-  GList        *p;
-  GHashTable *hash;
+  manager = udisks_client_get_object_manager (self->priv->client);
+  objects = g_dbus_object_manager_get_objects (manager);
+  added_drives = g_hash_table_new (g_str_hash, g_str_equal);
```

Comment:
Nitpick: I’d always use `g_hash_table_new_full()` instead, and pass `NULL` for the `GDestroyNotify` arguments to make it more explicit that the stored keys and values do not need to be freed.